Unresolved symbol when pickling triggered by inlining def that uses a type variable in summonFrom

## minimized code

```Scala
object Boom {
  import scala.compiletime._
  trait Fail[A <: Int, B <: Int]

  erased inline given fail[X <: Int, Y <: Int]: Fail[X, Y] = {
     scala.compiletime.summonFrom {
       case t: Fail[X, y] if constValue[y] < constValue[Y] => ???
    }
  }

  val a: Int = 1
  given ev1: Fail[a.type, 2] = null

  summon[Fail[a.type, 3]]
}
```


<details>
<summary>
assertion failed: unresolved symbols: type y(line 6) when pickling /testdotty/src/main/scala/boom.scala
</summary>

```scala
java.lang.AssertionError: assertion failed: unresolved symbols: type y(line 6) when pickling /testdotty/src/main/scala/boom.scala
dotty.DottyPredef$.assertFail(DottyPredef.scala:17)
dotty.tools.dotc.core.tasty.TreePickler.pickle(TreePickler.scala:704)
dotty.tools.dotc.transform.Pickler.run$$anonfun$10$$anonfun$8(Pickler.scala:63)
dotty.runtime.function.JProcedure1.apply(JProcedure1.java:15)
dotty.runtime.function.JProcedure1.apply(JProcedure1.java:10)
scala.collection.immutable.List.foreach(List.scala:305)
dotty.tools.dotc.transform.Pickler.run$$anonfun$2(Pickler.scala:87)
dotty.runtime.function.JProcedure1.apply(JProcedure1.java:15)
dotty.runtime.function.JProcedure1.apply(JProcedure1.java:10)
scala.collection.immutable.List.foreach(List.scala:305)
dotty.tools.dotc.transform.Pickler.run(Pickler.scala:87)
dotty.tools.dotc.core.Phases$Phase.runOn$$anonfun$1(Phases.scala:315)
scala.collection.immutable.List.map(List.scala:223)
dotty.tools.dotc.core.Phases$Phase.runOn(Phases.scala:316)
dotty.tools.dotc.transform.Pickler.runOn(Pickler.scala:91)
dotty.tools.dotc.Run.runPhases$4$$anonfun$4(Run.scala:161)
dotty.runtime.function.JProcedure1.apply(JProcedure1.java:15)
dotty.runtime.function.JProcedure1.apply(JProcedure1.java:10)
scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
dotty.tools.dotc.Run.runPhases$5(Run.scala:171)
dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:179)
dotty.runtime.function.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:65)
dotty.tools.dotc.Run.compileUnits(Run.scala:186)
dotty.tools.dotc.Run.compileSources(Run.scala:123)
dotty.tools.dotc.Run.compile(Run.scala:106)
dotty.tools.dotc.Driver.doCompile(Driver.scala:36)
dotty.tools.dotc.Driver.process(Driver.scala:189)
dotty.tools.dotc.Main.process(Main.scala)
```
</details>

--- a/compiler/src/dotty/tools/dotc/typer/Inliner.scala
+++ b/compiler/src/dotty/tools/dotc/typer/Inliner.scala
@@ -1035,7 +1035,7 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(implicit ctx: Context) {
               case _ => false
             }
           }
-          if (guardOK) Some((caseBindings, cdef.body.subst(from, to)))
+          if (guardOK) Some((caseBindings.map(_.subst(from, to)), cdef.body.subst(from, to)))
           else None
         }
         else None
@@ -1135,7 +1135,6 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(implicit ctx: Context) {
                 }
               case _ => rhs0
             }
-
             val (usedBindings, rhs2) = dropUnusedDefs(caseBindings, rhs1)
             val rhs = seq(usedBindings, rhs2)
             inlining.println(i"""--- reduce:

--- a/tests/pos/i7878.scala
+++ b/tests/pos/i7878.scala
@@ -1,0 +1,15 @@
+object Boom {
+  import scala.compiletime._
+  trait Fail[A <: Int, B <: Int]
+
+  erased inline given fail[X <: Int, Y <: Int]: Fail[X, Y] = {
+     scala.compiletime.summonFrom {
+       case t: Fail[X, y] if constValue[y] < constValue[Y] => ???
+    }
+  }
+
+  val a: Int = 1
+  given ev1: Fail[a.type, 2] = null
+
+  summon[Fail[a.type, 3]]
+}

